cli/mender-artifact/validate.go

```diff
@@ -59,6 +59,10 @@ func validate(art io.Reader, key []byte) error {
 		Log.Debug("key was specified but no digital signature was found")
 		return ErrInvalidSignature
 	}
+	if !keyIsSpecified && ar.IsSigned {
+		Log.Debug("no key was specified for a signed Artifact validation")
+		return ErrInvalidSignature
```

Comment:
I think all three of these conditions should have separate errors. And IMHO they should be more descriptive than currently, like "Missing key", "Missing signature" and "Signature is invalid", or something like that.

Reply:
I completely agree :smiley: 

My only excuse is that it was already simplified here and in the tests, and I decided on not modifying the existing design... However, I will re-work it now with your encouragement.

Reply:
I reworked it a bit. Let me know what you think now.